src/timeline/js/directives/ruler.js

```diff
@@ -144,6 +143,8 @@ App.directive("tlRuler", function ($timeout) {
           });
         });
 
+        drawTicks();
```

Comment:
I don't have any issues with refactoring the code into a method (i.e. drawTicks()) as you have done... and invoking it "on scroll" and "on scope.watch". I think this is a relatively simple fix of a very big performance issue (especially when zoomed into the timeline, causing a HUGE timeline pixel width).

Reply:
Trouble is, "on scroll" Only recieves scrolling when the mouse is over the ruler. And the tlBody or tlScrollableTracks doesn't have access to drawTicks where it is. Not sure what scope to put this function in so it can be called elsewhere in the webview